commit ae40d99663d8b2968176723f074872489cea7468 (refs/changes/91/2091/2)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-06-14T13:54:56-07:00 (2 years, 4 months ago)
    
    joyent/node-krill#10 createPredicate throws an error with erroneous message if type of field is not string, number or boolean

diff --git a/lib/krill.js b/lib/krill.js
index b390428..4750531 100644
--- a/lib/krill.js
+++ b/lib/krill.js
@@ -462,7 +462,7 @@ function krillPrimValidateRel(pred, key)
 	    typeof (constant) != 'string' &&
 	    typeof (constant) != 'boolean')
 		throw (new VError('predicate %j: field "%s" is not a string, ' +
-		    'number, or boolean', pred, key));
+		    'number, or boolean', pred, field));
 }
 
 /*
diff --git a/package.json b/package.json
index dbcb356..f14b873 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "krill",
 	"description": "simple boolean filter language with support for C, DTrace and LDAP output",
-	"version": "1.0.0",
+	"version": "1.0.1",
 	"license": "MIT",
 	"contributors": [
 	    "Robert Mustacchi <rm@joyent.com>",
diff --git a/tests/tst.invalid.js b/tests/tst.invalid.js
index 753e74e..6f67f20 100644
--- a/tests/tst.invalid.js
+++ b/tests/tst.invalid.js
@@ -36,7 +36,9 @@ var invalids = [
     [ { 'lt': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
     [ { 'le': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
     [ { 'ge': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
-    [ { 'gt': [ 's', '3' ] },	/cannot be applied to fields of type/, types ]
+    [ { 'gt': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
+    [ { 'eq': [ 'foo', null ] },
+        /predicate { eq: \[ 'foo', null \] }: field "foo" is not a string, number, or boolean/ ]
     /* END JSSTYLED */
 ];
 
